src/core/util/xlio_stats.h

```diff
@@ -225,44 +225,51 @@ typedef struct socket_listen_counters {
     }
 } socket_listen_counters_t;
 
-typedef struct socket_stats_t {
-    int fd;
-    uint32_t inode;
-    uint32_t tcp_state; // enum tcp_state
-    uint8_t socket_type; // SOCK_STREAM, SOCK_DGRAM, ...
-    bool padding1;
-    sa_family_t sa_family;
-    bool b_is_offloaded;
-    bool b_blocking;
-    bool b_mc_loop;
-    bool padding2;
-    in_port_t bound_port;
-    in_port_t connected_port;
-    ip_address bound_if;
-    ip_address connected_ip;
-    ip_address mc_tx_if;
-    pid_t threadid_last_rx;
-    pid_t threadid_last_tx;
-    uint32_t n_rx_ready_pkt_count;
-    uint32_t n_rx_ready_byte_limit;
+struct socket_stats_t {
+    // Data Path
+
+    union {
+        uint64_t n_tx_ready_byte_count;
+        socket_stats_t *_next_stat;
```

Comment:
AFAIU, `_next_stat` is used only to keep a stat object in the global free list. If yes, I'd suggest to move it to the cold zone (for example, to the end on the structure). Extra 8 cold bytes won't make an impact, but this will improve readability.

Reply:
Done